lib/spotify/types.ts

```diff
@@ -25,3 +25,25 @@ export function isUserInfoResponse(res: any): res is UserInfoResponse {
     'uri' in res
   );
 }
+
+type TopSongs = {
+  'name': string,
+  'href': string, 
+  'uri': string, 
+  'popularity': number, 
+  'artists': Array<string>,
+};
+
+export type TopSongsResponse = {
+  [key: string]: TopSongs,
```

Comment:
nit: this is more descriptive

```suggestion
  [id: string]: TopSongs,
```